src/groupeddataframe/grouping.jl

```diff
@@ -412,28 +412,38 @@ true
 """
 Base.keys(gd::GroupedDataFrame) = GroupKeys(gd)
 
-# Index with GroupKey
-function Base.getindex(gd::GroupedDataFrame, key::GroupKey)
-    gd === parent(key) && return gd[getfield(key, :idx)]
-    throw(ErrorException("Cannot use a GroupKey to index a GroupedDataFrame other than the one it was derived from."))
+
+# The allowed key types for dictionary-like indexing
+const AbstractGroupKey = Union{GroupKey, Tuple, NamedTuple}
+
+
+# Find the integer index of a group given its key
+function _findgroup(gd::GroupedDataFrame, key::GroupKey, strict::Val{S}=Val{true}()) where {S}
```

Comment:
Why do you use `strict` parameter here? It is not used later. I would just drop it and assume that we are always strict.
Also with constant propagation the `Val` trick in practice is not needed now (it was needed in early versions of Julia that had much less efficient compiler).